typescript/cli/src/status/message.ts

```diff
@@ -44,22 +44,27 @@ export async function checkMessageStatus({
   );
 
   let dispatchedReceipt: TransactionReceipt;
-  if (!dispatchTx) {
+
+  if (dispatchTx) {
+    dispatchedReceipt = await context.multiProvider
+      .getProvider(origin)
+      .getTransactionReceipt(dispatchTx);
+  } else {
     try {
       dispatchedReceipt = await core.getDispatchTx(origin, messageId);
     } catch (e) {
       logRed(`Failed to infer dispatch transaction for message ${messageId}`);
+
+      dispatchTx = await input({
+        message: 'Provide dispatch transaction hash',
+      });
+      dispatchedReceipt = await context.multiProvider
+        .getProvider(origin)
+        .getTransactionReceipt(dispatchTx);
     }
```

Comment:
I know you were fixing a bug but this doesnt make it much clearer

Reply:
Really, imo its more clear. What do you think would make it more clear?

Reply:
not having the receipt fetching in two places

Reply:
I would say its clear, but might feel duplicative, but IMO they are different code paths. Happy to hear suggestions as to how to make this clearer